### PR TITLE
Add mda-xdrlib and update pyedr and panedr

### PR DESCRIPTION
**Pull request by RMeli**
Add mda-xdrlib and update pyedr and panedr.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyedr/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class PyPyedr(PythonPackage):
 
     maintainers("RMeli")
 
+    version("0.7.2", sha256="8a02b7d94f97f99083c489568f7816ee8ed37e2efca0c1ba3a2e4b83e932d5b9")
     version("0.7.1", sha256="ad7ccdeb739399acd11a25f2d2413ebb46a54223059a2b902ac604d29fabd767")
 
     depends_on("python@3.6:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pyedr/package.py]: Optional: Spack only supports `python@3.7:` anyway, so you can remove this line.

File: var/spack/repos/builtin/packages/py-mda-xdrlib/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    maintainers("RMeli")
+
+    version("0.2.0", sha256="f26f7158a83c32b96d15b530fce2cbc1190c4b7024e41faa4ab3e3db74e272af")
+
+    depends_on("python@3.7:3.11", type=("build", "run"))
+    depends_on("py-wheel", type="build")
+
+    depends_on("py-setuptools@61.2:", type="build")
+    depends_on("py-tomli", type="build")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-mda-xdrlib/package.py]:
```suggestion
    depends_on("py-tomli", when="^python@:3.10", type="build")
```

File: var/spack/repos/builtin/packages/py-mda-xdrlib/package.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    homepage = "https://github.com/MDAnalysis/mda-xdrlib"
+    pypi = "mda_xdrlib/mda_xdrlib-0.2.0.tar.gz"
+
+    maintainers("RMeli")
+
+    version("0.2.0", sha256="f26f7158a83c32b96d15b530fce2cbc1190c4b7024e41faa4ab3e3db74e272af")
+
+    depends_on("python@3.7:3.11", type=("build", "run"))
+    depends_on("py-wheel", type="build")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-mda-xdrlib/package.py]: `py-wheel` is added as dependency by default. You would only need to specify it again if a certain version is needed.

File: var/spack/repos/builtin/packages/py-mda-xdrlib/package.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    """A stand-alone XDRLIB module extracted from CPython 3.10.8"""
+
+    homepage = "https://github.com/MDAnalysis/mda-xdrlib"
+    pypi = "mda_xdrlib/mda_xdrlib-0.2.0.tar.gz"
+
+    maintainers("RMeli")
+
+    version("0.2.0", sha256="f26f7158a83c32b96d15b530fce2cbc1190c4b7024e41faa4ab3e3db74e272af")
+
+    depends_on("python@3.7:3.11", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-mda-xdrlib/package.py]: Can you add a comment where this upper restriction comes from?

**Comment by RMeli** [comment]
Thanks @manuelakuhn for the review, I think I addressed all the issues.

File: var/spack/repos/builtin/packages/py-mda-xdrlib/package.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    homepage = "https://github.com/MDAnalysis/mda-xdrlib"
+    pypi = "mda_xdrlib/mda_xdrlib-0.2.0.tar.gz"
+
+    maintainers("RMeli")
+
+    version("0.2.0", sha256="f26f7158a83c32b96d15b530fce2cbc1190c4b7024e41faa4ab3e3db74e272af")
+
+    # See pyproject.toml for restriction on Python versions
+    depends_on("python@:3.10", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mda-xdrlib/package.py]: I don't see this restriction in pyproject.toml

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mda-xdrlib/package.py (hunk 5 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-mda-xdrlib/package.py]: That's right. The `python < 3.11` pin for the build-system is only there for `tomli`, but that is already taken care of.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**